### PR TITLE
Melody96 QMK Support

### PR DESCRIPTION
**Pull request by kaylanm**
Created based on the JSON here: https://drive.google.com/file/d/1oJmSfjgs_DugsvloO9VaO5KWvwS4T2OX/view provided in the GH thread for the GB here: https://geekhack.org/index.php?topic=93614.0 and verified on actual hardware.

### CODE REVIEW

File: keyboards/melody96/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#include "melody96.h"
```
Comment (drashna) [inline keyboards/melody96/keymaps/default/keymap.c]: Please replace this with `#include QMK_KEYBOARD_H`

File: keyboards/melody96/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#define _______ KC_TRNS
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (drashna) [inline keyboards/melody96/keymaps/default/keymap.c]: Please replace all of the `KEYMAP`'s with `LAYOUT`

File: keyboards/melody96/melody96.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#ifndef KB_H
+#define KB_H
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (drashna) [inline keyboards/melody96/melody96.h]: Please change the `KEYMAP` to `LAYOUT`

File: keyboards/melody96/melody96.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#ifndef KB_H
```
Comment (drashna) [inline keyboards/melody96/melody96.h]: (Optional) Could you change this to `MELODY_H` or the like, so it's not generic?

File: keyboards/melody96/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# YMDK Melody96
+
+![melody96](https://i.imgur.com/9o0a6XF.png)
+
+TKL Keyboard from KBDFans
+
+Keyboard Maintainer: [superuser41](https://github.com/kaylanm)
```
Comment (drashna) [inline keyboards/melody96/readme.md]: Make sure these three lines have a double space at the end of them.

File: keyboards/melody96/rules.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/melody96/rules.mk]: Do we know what bootloader is used here? 

Or if it's a Pro Micro/Teensy/etc?

Comment (drashna) [reply keyboards/melody96/rules.mk]: Ah, amtel dfu.  If that's the case, we don't need the bootloader size, but it doesn't hurt (it's redefined later, based on the amtel dfu stuff)

**Comment by drashna** [comment]
Awesome!  Thanks!

**Comment by kaylanm** [comment]
Thank you!

**Pull request merged**